tool/src/org/antlr/v4/codegen/Target.java

```diff
@@ -126,7 +126,7 @@ protected void genFile(Grammar g, ST outputFileST, String fileName)
 	 *  to a token type in the generated code.
 	 */
 	public String getTokenTypeAsTargetLabel(Grammar g, int ttype) {
-		String name = g.getTokenName(ttype);
+	    String name = this.escapeIfNeeded(g.getTokenName(ttype));
```

Comment:
I'm surprised this used to work at all. Did we not need the escape previously?

Reply:
Was this not being escaped before? Hmm... Seems like without the escape none of the reserved words would have been rewritten.

Reply:
@parrt  Because all the reserve words so far for CppTargets are lower case and thus are never expected to appear as token definition as per antlr specification (rulenames in lower case and tokens in uppercase).

Reply:
oh ok, got it!